Allow input, output while reading rule from file

Jollyfant opened the pull request:
I'm not sure this is the intended behaviour but I have a static rule body without `INPUT` or `OUTPUT` and wish to supply this dynamically. Before this commit, it was either possible to 1) read everything from file or 2) supply everything (having to pass a body in addition to I/O).

trel commented:
Hi @Jollyfant,

Can you give a more concrete example of what this now allows that didn't work before?

(Double bonus points if it's in the form of a test)

Jollyfant commented:
When passing a rule body to the function it has to include both the rule body, the `INPUT` and `OUTPUT`.

I want to pass a rule body e.g. 

    eudatPidSingleCheck2 {

      EUDATSearchPID(*path, *existing_pid)
      if(*existing_pid == "empty") {
        EUDATCreatePID(*parent_pid, *path, *ror, *fio, *fixed, *newPID);
        writeLine("stdout","PID-new: *newPID");
      }
      else {
        writeLine("stdout","PID-existing: *existing_pid");
      }

    }

and supply the I/O through the python library `params` and `output`

    INPUT *path="null",*pid="null",*parent_pid="none",*ror="none",*fio="none",*fixed="False"
    OUTPUT ruleExecOut

Otherwise we have to a write a temporary file with a dynamic `INPUT`, load the rule from file and execute it. Alternative, we could add the body to the function call but I rather have it statically on disk.

trel commented:
We're running this through CI.

trel commented:
These two tests failed...
- irods.test.rule_test.TestRule.test_add_metadata_from_rule_file
- irods.test.rule_test.TestRule.test_retrieve_std_streams_from_rule

Both with code `-1211000`...

```
$ ierror 1211000
irods error: -1211000 RE_UNABLE_TO_READ_LOCAL_VAR
```

We'll look into how to get this to work.

trel commented:
Running latest commits through CI.

trel commented:
Passed!

trel commented:
Please squash to a single commit - and we'll get it merged.

Jollyfant commented:
Squashed!

trel commented:
Excellent.
